File: src/javascript/5kyu/rot13.js
// https://www.codewars.com/kata/52223df9e8f98c7aa7000062

function rot13(str) {
    const up = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split('');
    const lo = "abcdefghijklmnopqrstuvwxyz".split('');

    return str
        .split('')
        .map(l => {
            if (up.indexOf(l) >= 0) {
                return up[(up.indexOf(l) + 13) % 26];
            } else if (lo.indexOf(l) >= 0) {
                return lo[(lo.indexOf(l) + 13) % 26];
            } else {
                return l;
            }
        })
        .join('')
}

File: src/python/5kyu/rot13.py
# https://www.codewars.com/kata/530e15517bc88ac656000716

import string
import operator
from codecs import encode as _dont_use_this_


def rotateChar(c):
    lower = list(string.ascii_lowercase)
    upper = list(string.ascii_uppercase)
    if c in lower:
        return lower[(operator.indexOf(lower, c) + 13) % 26]
    elif c in upper:
        return upper[(operator.indexOf(upper, c) + 13) % 26]
    else:
        return c


def rot13(message):
    return ''.join(list(map(lambda x: rotateChar(x), list(message))))
